awsplugins/ec2/ec2_test.go

```diff
@@ -0,0 +1,122 @@
+package ec2
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const metadata = `{
+  "availabilityZone" : "us-west-2a",
```

Comment:
I think the previous version with a more representative doc was better, it does add a bit more test coverage by making sure unexpected keys don't break anything